src/library/scala/collection/mutable/HashMap.scala

```diff
@@ -161,7 +161,9 @@ class HashMap[K, V](initialCapacity: Int, loadFactor: Double)
           else table(indexedHash) = foundNode.next
           contentSize -= 1
 
-        case (None, Some(value)) => put0(key, value, false, hash, indexedHash)
+        case (None, Some(value)) =>
+          if(contentSize + 1 >= threshold) growTable(table.length * 2)
+          put0(key, value, false, hash, indexedHash)
```

Comment:
Isn't `indexedHash` invalidated once you grow the table?

Reply:
Yeah, just noticed that.

Reply:
So it's not possible to never re-compute the hash. If we're at the threshold the implementation pretty much falls back to the original one. Anyway, this could also just call `put0` but then we're doing threshold check twice so it's either that or DRY.

Reply:
Sorry, what I meant is *compute the index*, hash stays the same.